### DISCUSSION TITLE
7 second delay on playback start

### DISCUSSION THREAD

**Discussion by codydubat** [dtcooper/raspotify#620] [General]
Hi everyone,
I noticed that since the past 2 updates, there is an approximately 7 second delay from the moment I hit the play button on any other device, until raspotify starts playing the song.
This happens not only when you load a new song but also when you hit pause (pauses in 1sec) and then hit play again.
Is anyone else having this issue, or is it just on my side? (using raspberry 3b+)

**Comment by JasonLG1979** [comment]
More than likely duplicate of https://github.com/dtcooper/raspotify/issues/619

**Comment by JasonLG1979** [comment]
So I've found a device (My monitor's built in DAC via HDMI) with which I can reproduce a couple different issues that cause a delay.

One issue is that calculating the buffer and period sizes took A REALLY LONG time (5 to 10 secs). Which I believe I've fixed by tightening the range in which buffer/period sizes are tested.

Another issue with the same device is that it's default buffer and period sizes are HUGE. So in the event that calculating the buffer and period sizes failed for whatever reason you'd stuck with that huge default buffer which could lead to basically waiting for the whole song to completely download before playback begins. I believe I've fixed that also by adjusting the min and max tested buffer and period ranges to be more reasonable.

I've pushed a new version see if that helps. If not we will need to do some further debugging.

**Reply by codydubat** [reply]
That's great to hear! Thanks a lot for your efforts, I will try to pull the changes, test and let you know.

**Reply by codydubat** [reply]
@JasonLG1979  I just tested the latest version and it's waay better! There is still a bit of delay, like 2-3 seconds (can't remember if it was like this also before), but a definite improvement! Thanks, you rock!

**Reply by JasonLG1979** [reply]
I streamlined everything even more and released a new version. That's about as fast as it can get.

**Reply by codydubat** [reply]
Wow, now it's almost instant, as fast as local playback, indeed I don't think it can go any faster 😆. Thank you for the efforts!
